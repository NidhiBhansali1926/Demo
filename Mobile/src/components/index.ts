export { default as Header } from './Header';
export { default as AppButton } from './AppButton';
export { default as AppBackground } from './AppBackground';
