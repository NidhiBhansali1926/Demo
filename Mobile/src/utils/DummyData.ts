export const dummyData = {
  token: 'axyruriurirfjfnfmniriryoabcdopeoriro',
};
