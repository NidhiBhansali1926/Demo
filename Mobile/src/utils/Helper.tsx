import { NavigationProp } from '@react-navigation/native';
import React, { ReactElement } from 'react';
import { Provider } from 'react-redux';
import { create } from 'react-test-renderer';

import { store } from '../store/index';

import { AuthParamsList } from '@typing/navigation';

export const renderScreen = (screen: ReactElement) => {
  const myScreen = create(<Provider store={store}>{screen}</Provider>);
  return myScreen;
};

export const getPropsWithNavigation = (props?: unknown, navigationPropExtension?: Partial<NavigationProp<AuthParamsList>>) => ({
  ...props,
  navigation: {
    navigate: jest.fn(),
    goBack: jest.fn(),
    replace: jest.fn(),
    dispatch: jest.fn(),
    getState: jest.fn(() => {}),
    isFocused: () => true,
    addListener: jest.fn().mockImplementation((event, callback) => {
      callback();
      return {
        remove: jest.fn(),
      };
    }),
    ...navigationPropExtension,
  },
});
