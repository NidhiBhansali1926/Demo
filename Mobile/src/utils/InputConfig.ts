export enum KeyboardType {
  default = 'default',
  numeric = 'numeric',
  emailAddress = 'email-address',
  phonePad = 'phone-pad',
}

export enum AutoCapitalize {
  characters = 'characters',
  words = 'words',
  sentences = 'sentences',
  none = 'none',
}

export enum ReturnKeyType {
  next = 'next',
  done = 'done',
}
export const InputConfig = {};
