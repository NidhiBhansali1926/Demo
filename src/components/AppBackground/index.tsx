import { LinearGradient } from 'expo-linear-gradient';
import React from 'react';
import { ImageBackground, ImageBackgroundProps } from 'react-native';
import SafeAreaView, { ForceInsetProp } from 'react-native-safe-area-view';

import { useTheme } from '@hooks';

interface AppBackgroundProps extends ImageBackgroundProps {
  children: React.ReactNode;
  colors?: string[];
  location?: number[];
  overlayColor?: string;
}

const AppBackground = ({ children, colors, location, overlayColor, ...imageBackgroundProps }: AppBackgroundProps) => {
  const { Layout, colorThemeVariable } = useTheme();

  return (
    <ImageBackground style={Layout.fill} {...imageBackgroundProps}>
      <LinearGradient
        colors={colors ?? [colorThemeVariable.transparent, colorThemeVariable.transparent]}
        locations={location ?? [0, 1]}
        style={Layout.fill}
      >
        <SafeAreaView
          style={[Layout.fill, { backgroundColor: overlayColor ?? colorThemeVariable.transparent }]}
          forceInset={Layout.safeareaViewInsent as ForceInsetProp}
        >
          {children}
        </SafeAreaView>
      </LinearGradient>
    </ImageBackground>
  );
};

export default AppBackground;
