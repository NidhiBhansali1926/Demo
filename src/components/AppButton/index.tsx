import React from 'react';
import { ActivityIndicator, ImageSourcePropType, ImageStyle, Pressable, StyleProp, Text, TextStyle, ViewStyle } from 'react-native';

import styles from './styles';

import { useTheme } from '@hooks';

interface ButtonProps {
  name?: string;
  containerStyle?: StyleProp<ViewStyle>;
  textStyle?: StyleProp<TextStyle>;
  icon?: ImageSourcePropType;
  iconStyle?: ImageStyle;
  handleClick?: () => void;
  isLoading?: boolean;
  disabled?: boolean;
}

const AppButton = ({ name, containerStyle, textStyle, handleClick, isLoading = false, disabled }: ButtonProps) => {
  const { Fonts, Layout, colorThemeVariable } = useTheme();

  return (
    <Pressable
      disabled={disabled || isLoading}
      style={[
        styles.buttonContainer,
        Layout.center,
        { backgroundColor: colorThemeVariable.primary },
        isLoading || disabled ? { backgroundColor: colorThemeVariable.disabledPrimary } : {},
        containerStyle,
      ]}
      onPress={handleClick}
    >
      {isLoading && <ActivityIndicator size='small' style={Layout.absoluteFillObject} color={colorThemeVariable.background} />}
      <Text
        style={[
          Fonts.textNunitoMedium,
          Fonts.textCenter,
          Fonts.textSmallPlus,
          Fonts.textLineHeightDefault,
          { color: colorThemeVariable.title },
          textStyle,
        ]}
      >
        {isLoading ? '' : name}
      </Text>
    </Pressable>
  );
};

export default AppButton;
