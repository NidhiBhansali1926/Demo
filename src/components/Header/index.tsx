import React from 'react';
import { useTranslation } from 'react-i18next';
import { Pressable, PressableProps, Text, View } from 'react-native';

import styles from './styles';

import { useTheme } from '@hooks';

import { ArrowSVG } from '@svg';
import Methods from '@utils/Method';

export interface MenuBarProps {
  handleOnPressBackIcon: PressableProps['onPress'];
}

const Header = React.memo(({ handleOnPressBackIcon }: MenuBarProps) => {
  const { Fonts, Layout, Colors } = useTheme();
  const { t } = useTranslation(['elevateAdventure']);

  return (
    <View style={[styles.container, Layout.row, Layout.justifyContentBetween]}>
      <Pressable hitSlop={5} onPress={handleOnPressBackIcon}>
        <ArrowSVG />
      </Pressable>
      <Pressable onPress={Methods.inDevelopmentAlert}>
        <Text style={[Fonts.textSmallPlus, Fonts.textNunitoMedium, { color: Colors.commonThemeColor.black }]}>
          {t('elevateAdventure:saveForLater')}
        </Text>
      </Pressable>
    </View>
  );
});

export default Header;
