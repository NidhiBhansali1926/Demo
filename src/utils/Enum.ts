import { KeyboardHideEventType } from '../../typing/common';

export const typeOfKeyboardEvent: { willHide: KeyboardHideEventType; didHide: KeyboardHideEventType } = {
  willHide: 'keyboardWillHide',
  didHide: 'keyboardDidHide',
};

// Request headers
export enum Headers {
  Authorization = 'authorization',
  ContentType = 'Content-Type',
}
