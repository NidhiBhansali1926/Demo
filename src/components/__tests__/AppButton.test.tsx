import React from 'react';

import { AppButton } from '@src/components';
import { getPropsWithNavigation, renderScreen } from '@src/utils/Helper';

describe('Snapshot Test for Rendering AppButton Component', () => {
  test('Snapshot of AppButton Component with Navigation Props', () => {
    const props = getPropsWithNavigation();
    const buttonProps = {
      ...props,
      name: 'buttonText',
      isLoading: false,
      disabled: false,
      handleClick: jest.fn(),
    };
    const appButtonSnapshot = renderScreen(<AppButton {...buttonProps} />).toJSON();
    expect(appButtonSnapshot).toMatchSnapshot();
  });
});
