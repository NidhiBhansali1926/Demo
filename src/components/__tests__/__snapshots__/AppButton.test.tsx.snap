// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Test for Rendering AppButton Component Snapshot of AppButton Component with Navigation Props 1`] = `
<AppButton
  disabled={false}
  handleClick={[MockFunction]}
  isLoading={false}
  name="buttonText"
  navigation={
    {
      "addListener": [MockFunction],
      "dispatch": [MockFunction],
      "getState": [MockFunction],
      "goBack": [MockFunction],
      "isFocused": [Function],
      "navigate": [MockFunction],
      "replace": [MockFunction],
    }
  }
/>
`;
